six/modules/c++/six.sidd/tests/test_sidd_blocking.cpp

```diff
@@ -0,0 +1,194 @@
+        const std::string& siddPathname)
+{
+    const std::string siddCreator =
+            getProgramPathname(installPathname, "test_create_sidd_from_mem");
+    sys::Exec createSiddCommand(siddCreator + " --lut Mono --multipleSegments "
+            + siddPathname);
+    createSiddCommand.run();
+}
+
+void roundTripAndBlock(const std::string& installPathname,
+        const std::string& inputPathname,
+        const std::string& outputPathname)
+{
+    const std::string roundTripProgram =
+            getProgramPathname(installPathname, "round_trip_six");
+    sys::Exec roundTripCommand(roundTripProgram + " --retainDateTime" +
+            " --size 5000 --rowsPerBlock 10 --colsPerBlock 10 " +
+            inputPathname + " " + outputPathname);
+    roundTripCommand.run();
+}
+
+bool checkBlocking(const std::string& originalPathname,
+        const std::string& convertedPathname)
+{
+    six::NITFReadControl originalReader;
+    originalReader.load(originalPathname);
+    six::NITFReadControl convertedReader;
+    convertedReader.load(convertedPathname);
+
+    mem::SharedPtr<six::Container> originalContainer =
+            originalReader.getContainer();
+
+    six::Region originalRegion;
+    originalRegion.setStartRow(0);
+    originalRegion.setStartCol(0);
+
+    six::Region convertedRegion;
+    convertedRegion.setStartRow(0);
+    convertedRegion.setStartCol(0);
+
+    const size_t numRows = originalContainer->getData(0)->getNumRows();
+    const size_t numCols = originalContainer->getData(0)->getNumCols();
+    const size_t bytesPerPixel =
+            originalContainer->getData(0)->getNumBytesPerPixel();
+    const size_t bufferSize = numRows * numCols * bytesPerPixel;
+
+    mem::ScopedArray<six::UByte> originalBuffer(
+            originalReader.interleaved(originalRegion, 0));
+    mem::ScopedArray<six::UByte> convertedBuffer(
+            convertedReader.interleaved(convertedRegion, 0));
+    for (size_t jj = 0; jj < bufferSize; ++jj)
+    {
+        if (originalBuffer[jj] != convertedBuffer[jj])
+        {
+            std::cerr << "Error with round trip\n";
+            return false;
+        }
+    }
+    std::cerr << "Round trip succeeded\n";
+    return true;
+}
+}
+
+int main(int argc, char** argv)
+{
+    if (argc != 2)
+    {
+        std::cerr << "Usage: " << argv[0] << " <installPathname>\n";
+        return 1;
+    }
+    try
+    {
+        six::XMLControlFactory::getInstance().addCreator(
+            six::DataType::DERIVED,
+            new six::XMLControlCreatorT<
+            six::sidd::DerivedXMLControl>());
+
+        const std::string installPathname(argv[1]);
+        TempFileWithExtension multiImageSidd(".nitf");
+        TempFileWithExtension roundTrippedSidd(".nitf");
+        getMultiImageSIDD(installPathname, multiImageSidd.pathname());
+        roundTripAndBlock(installPathname,
+                multiImageSidd.pathname(), roundTrippedSidd.pathname());
+
+        if (checkBlocking(multiImageSidd.pathname(),
```

Comment:
This feels a little hokey to be calling other C++ applications inside C++ and then within Python calling this but I get why you did it this way and it does make the test overall simple... the alternative would maybe be to make function calls that do 98% of the implementation of round_trip_six and test_create_sidd_from_mem but it's probably not worth the trouble.